MoreLinq/Fold.cs

```diff
@@ -96,7 +100,7 @@ static Exception OnFolderSourceSizeError(int cmp, int count)
             var message = cmp < 0
                         ? "Sequence contains too few elements when exactly {0} {1} expected."
                         : "Sequence contains too many elements when exactly {0} {1} expected.";
```

Comment:
See comment on `AssertCount.cs`

Reply:
[Same answer](https://github.com/morelinq/MoreLINQ/pull/924#discussion_r1070661519) as for [comment on `AssertCount.cs`](https://github.com/morelinq/MoreLINQ/pull/924#discussion_r1070624334).